ibis/pyspark/timecontext.py

```diff
@@ -26,10 +26,40 @@ def filter_by_time_context(
 
     begin, end = timecontext
     if TIME_COL in df.columns:
-        return df.filter((F.col(TIME_COL) >= begin) & (F.col(TIME_COL) < end))
+        # for py3.8, underlying spark type converter calls utctimetuple()
+        # and will throw excpetion for Timestamp type.
+        return df.filter(
+            (F.col(TIME_COL) >= begin.to_pydatetime())
+            & (F.col(TIME_COL) < end.to_pydatetime())
+        )
     else:
         raise com.TranslationError(
             "'time' column missing in Dataframe {}."
             "To use time context, a Timestamp column name 'time' must"
             "present in the table. ".format(df)
         )
+
+
+def combine_time_context(
+    timecontexts: List[TimeContext],
+) -> Optional[TimeContext]:
+    """ Return a combined time context of `timecontexts`
+
+    The combined time context starts from the earliest begin time
+    of `timecontexts`, and ends with the latest end time of `timecontexts`
+    The motivation is to generate a time context that is a superset
+    to all time contexts.
+
```

Comment:
add an Examples section if this is useful

Reply:
+1, I added an example